src/message.rs

```diff
@@ -0,0 +1,61 @@
+//! Message parsing
+//!
+//! Analagous to <http://netfilter.org/projects/libnetfilter_queue/doxygen/group__Parsing.html>
+
+use util::*;
+use ffi::*;
+pub use ffi::nfqnl_msg_packet_hdr as Header;
+
+pub struct Message<'a> {
+    pub raw: *mut nfgenmsg,
+    pub ptr: *mut nfq_data,
+    pub header: &'a Header
+}
+
+impl<'a> Drop for Message<'a> {
+    fn drop(&mut self) {}
+}
+
+impl<'a> Message<'a> {
+    pub fn new(raw: *mut nfgenmsg, ptr: *mut nfq_data) -> Message<'a> {
+        let header = unsafe {
+            let ptr = nfq_get_msg_packet_hdr(ptr);
+            as_ref(&ptr).unwrap()
+        };
+        Message {
+            raw: raw,
+            ptr: ptr,
+            header: header
+        }
+    }
+}
+
+pub mod verdict {
+    use libc::*;
+    use ffi::*;
+
+    pub enum Verdict {
+        Drop,
+        Accept,
+        Stolen,
+        Queue,
```

Comment:
Queue should prob be Queue(u16) as the verdict's target queue is encoded in the top 16 bytes